ts/components/messages/MessagesSearch.tsx

```diff
@@ -0,0 +1,145 @@
+import { fromNullable, none, Option, some } from "fp-ts/lib/Option";
+import * as pot from "italia-ts-commons/lib/pot";
+import { View } from "native-base";
+import React, { ComponentProps } from "react";
+import { StyleSheet } from "react-native";
+
+import { lexicallyOrderedMessagesStateSelector } from "../../store/reducers/entities/messages";
+import { MessageState } from "../../store/reducers/entities/messages/messagesById";
+import { ServicesByIdState } from "../../store/reducers/entities/services/servicesById";
+import { messageContainsText } from "../../utils/messages";
+import { serviceContainsText } from "../../utils/services";
+import MessageListComponent from "./MessageListComponent";
+
+const styles = StyleSheet.create({
+  listWrapper: {
+    flex: 1
+  }
+});
+
+type OwnProps = {
+  messagesState: ReturnType<typeof lexicallyOrderedMessagesStateSelector>;
+  searchText: string;
+  navigateToMessageDetail: (id: string) => void;
+};
+
+type Props = Pick<
+  ComponentProps<typeof MessageListComponent>,
+  "servicesById" | "paymentByRptId" | "onRefresh"
+> &
+  OwnProps;
+
+type State = {
+  lastMessagesState: ReturnType<typeof lexicallyOrderedMessagesStateSelector>;
+  lastSearchText: Option<string>;
+  filteredMessageStates: ReturnType<
+    typeof generateMessagesStateMatchingSearchTextArray
+  >;
+};
+
+/**
+ * Filter only the messages that matchs the searchText.
+ * The searchText is checked both in message and in service properties.
+ */
+const generateMessagesStateMatchingSearchTextArray = (
+  potMessagesState: pot.Pot<ReadonlyArray<MessageState>, string>,
+  servicesById: ServicesByIdState,
+  searchText: string
+): ReadonlyArray<MessageState> => {
+  return searchText === ""
```

Comment:
I think we should require at least a few characters for the search keyword, it doesn't really make sense to search for a keyword of less than 3 characters.

 ```suggestion
  return searchText.length < 3
```